lib/mirage.ml

```diff
@@ -2027,6 +1939,62 @@ type tracing = Tracing.t
 let mprof_trace ~size () =
   { Tracing.size }
 
+module Nocrypto_entropy = struct
+
+  (* XXX
+   * Nocrypto needs `mirage-entropy-xen` if compiled for xen.
+   * This is currently handled by always installing this library, regardless of
+   * usage of Nocrypto.
+   * As `libraries` is run after opam setup, it will correctly detect Nocrypto
+   * usage and inject appropriate deps for linkage, if needed.
+   * The `packages` function, unconditional installation of
+   * `mirage-entropy-xen`, and this comment, should be cleared once we have
+   * conditional dependencies support in opam.
+   * See https://github.com/mirage/mirage/pull/394 for discussion.
+   *)
+
+  module SS = StringSet
+
+  let remembering r f =
+    match !r with
+    | Some x -> x
+    | None   -> let x = f () in r := Some x ; x
+
+  let linkage_ = ref None
+
+  let linkage () =
+    match !linkage_ with
+    | None   -> failwith "Nocrypto_entropy: `linkage` queried before `libraries`"
+    | Some x -> x
+
+  let packages () =
+    match !mode with
+    | `Xen -> SS.singleton "mirage-entropy-xen"
```

Comment:
should we be explicit about the version? (such as `>=0.3.0`)!? previous `mirage-entropy-xen` will not work...